### PR TITLE
WIP: Add German (Germany) locale

### PR DESCRIPTION
**Pull request by Cogitri**
This is a WIP to add German localisation to Joplin. It's currently like ~35% done, however, I don't know electron apps too well, so I'll count on someone else to actually do the necessary changes to integrate it into Joplin ( which is required before this is merged, so that we can proofread  it in the actual application ).

### CODE REVIEW

**Comment by laurent22** [comment]
Thank you for the pull request. The PO file is all that's needed, I will then run a script that add the translation to all the applications automatically.

File: CliClient/locales/de_DE.po [hunk 1]
```diff
@@ -0,0 +1,957 @@
...
+msgid "Exit the application."
+msgstr "Verlasse das Programm."
+
+msgid "Delete the currently selected note or notebook."
+msgstr ""
+"Lösche die momentan ausgewählte Notiz oder das momentan ausgewählte "
+"Notizbuch."
+
+msgid "To delete a tag, untag the associated notes."
+msgstr "Um eine Markierung zu löschen, entmarkiere markierte Notizen."
```
Comment (kossmoboleat) [inline CliClient/locales/de_DE.po]: `entmarkiere markierte` is kind of clunky

File: CliClient/locales/de_DE.po [hunk 2]
```diff
@@ -0,0 +1,957 @@
...
+
+msgid "Please select the note or notebook to be deleted first."
+msgstr ""
+"Bitte wählen Sie zuert eine Notiz oder ein Notizbuch aus, das gelöscht "
+"werden soll."
+
+msgid "Set a to-do as completed / not completed"
+msgstr "Markiere ein ToDo as abgeschlossen / nicht abgeschlossen"
```
Comment (kossmoboleat) [inline CliClient/locales/de_DE.po]: Probably rather `To-Do`,  I think: https://dict.leo.org/forum/viewUnsolvedquery.php?idThread=538311&idForum=2&lang=en&lp=ende

File: CliClient/locales/de_DE.po [hunk 3]
```diff
@@ -0,0 +1,957 @@
...
+
+#, fuzzy
+msgid "[t]oggle [c]onsole between maximized/minimized/hidden/visible."
+msgstr ""
+"Schal[t]e das Terminal zwischen maximiert/minimiert/versteck und sichtbar um."
```
Comment (kossmoboleat) [inline CliClient/locales/de_DE.po]: `versteckt` and I think you could just keep the style without `und`

File: CliClient/locales/de_DE.po [hunk 4]
```diff
@@ -0,0 +1,957 @@
...
+msgid "[t]oggle note [m]etadata."
+msgstr "Schal[t]e Notiz-[M]etadata ein."
+
+#, fuzzy
+msgid "[M]ake a new [n]ote"
+msgstr "Erstelle eine neue Notiz"
+
+#, fuzzy
+msgid "[M]ake a new [t]odo"
+msgstr "Erstell ein neues ToDo"
```
Comment (kossmoboleat) [inline CliClient/locales/de_DE.po]: Erstelle

File: CliClient/locales/de_DE.po [hunk 5]
```diff
@@ -0,0 +1,957 @@
...
+
+#, fuzzy
+msgid "[M]ake a new note[b]ook"
+msgstr "Erstell ein neues Notiz[b]uch"
```
Comment (kossmoboleat) [inline CliClient/locales/de_DE.po]: Erstelle

File: CliClient/locales/de_DE.po [hunk 6]
```diff
@@ -0,0 +1,957 @@
...
+
+msgid ""
+"Gets or sets a config value. If [value] is not provided, it will show the "
+"value of [name]. If neither [name] nor [value] is provided, it will list the "
+"current configuration."
+msgstr ""
+
+msgid "Also displays unset and hidden config variables."
+msgstr ""
+"Zeige auch nicht angegebene oder versteckte Konfigurationsvariablen an."
```
Comment (kossmoboleat) [inline CliClient/locales/de_DE.po]: Should probably not be on an extra line?

File: CliClient/locales/de_DE.po [hunk 7]
```diff
@@ -0,0 +1,957 @@
...
+msgstr "Anstehender Alarm"
+
+#, javascript-format
+msgid "On %s: %s"
+msgstr "Auf %s: %s"
+
+msgid "There are currently no notes. Create one by clicking on the (+) button."
+msgstr ""
+"Momentan existieren noch keine Notizen. Erstelle eine indem Sie auf den (+) "
+"Knopf drücken."
```
Comment (kossmoboleat) [inline CliClient/locales/de_DE.po]: Does that work over several lines like this?

**Comment by Cogitri** [comment]
> Does that work over several lines like this?

Not sure, I want to do a first version with everything translated first and afterwards go on and see how it works in Joplin :)

Thanks for the feedback though :+1

-------- Original-Nachricht --------
An 2. Dez. 2017, 10:20, kossmoboleat schrieb:

File: CliClient/locales/de_DE.po (hunk 2 above)
Comment (Cogitri) [reply CliClient/locales/de_DE.po]: Yeah, To-Do sounds better, thanks :)

**Comment by Cogitri** [comment]
Almost all lines are translated now. Which script has to be run in order for me to be able to see the translation in action, @laurent22 ?

Also, would you mind looking over the translation again, once I did the above and looked over the translation again, @kossmoboleat ?

**Comment by laurent22** [comment]
Great, if you can build the app as described [here](https://github.com/laurent22/joplin/blob/master/BUILD.md), you can run the command `CliClients/build-translation.sh` which would add the translation to the three applications.

Alternatively, feel free to let me know which app you are using under which system, and I can create a beta version which you can check.

**Comment by Cogitri** [comment]
Hm, I've run `node Tools/build-translation.js`, but it seems like nothing happened. Would you mind providing me a build with the updated translation for linux?

**Comment by laurent22** [comment]
Hello, I don't have a Linux VM available right now (it's built automatically for official releases) but I've added the translation to master and built it. Please could you try again to build the app using the instructions here? https://github.com/laurent22/joplin/blob/master/BUILD.md

**Comment by Cogitri** [comment]
Thanks for that @laurent22 

I  guess the translation is now somewhat good to go. However, I couldn't come up with good translations for some 20 strings, would you mind to go over it, @kossmoboleat ? Maybe it'd be easier if we merged this PR and @kossmoboleat did his own though, it'd be kinda annoying if he had to do it via code reviews in this PR.

**Comment by laurent22** [comment]
The translation is now part of the latest release: https://github.com/laurent22/joplin/releases If you see anything that should be changed, feel free to update it here and I'll merge it for the next release.

**Comment by Cogitri** [comment]
I did some adjustments, should be good to go now @laurent22.
Also, although I'm cool with it, you may want to use `git format-patch` and `git am` to integrate patches to conserve the author etc :)

**Comment by laurent22** [comment]
Thanks for the update, and yes I should indeed have tried to keep attribution for the initial commit.

**Comment by laurent22** [comment]
So the final translation will be in v0.10.40+

**Pull request merged**